dask/array/tests/test_random.py

```diff
@@ -161,6 +161,11 @@ def test_random_all():
     da.random.standard_normal(size=5, chunks=3).compute()
     da.random.standard_t(2, size=5, chunks=3).compute()
 
+def test_array_broadcasting():
+    arr = np.arange(6).reshape((2,3))
+    assert da.random.poisson(arr, chunks=3).compute().shape == (2,3)
+    assert da.random.normal(arr, 2, chunks=3).compute().shape == (2,3)
```

Comment:
The user-supplied size (shape actually) is always respected - if given, only arrays of the same size (shape) will be broadcasted, the rest will just be passed as-is.

If no user-given size (shape) is found, then it checks if all arrays in the arguments are of the same size (shape) and uses if it's unique. If it's a mixed bag, then raises an error as it's ambiguous.

Which actually raises a problem - I'm not sure there are `np.random.*` functions that can use arguments that are not scalars and not of the intended size (i.e. when args arrays are of different sizes(shapes) than the user-given one).. So it may be slightly an overkill what I did there, as I have no direct way to test this.

As for the chunkshape, it behaves exactly as it would if you did something like so:
`da.ones((3,2), chunks=(300, 300, 300))`

I'll add some more tests though.